src/objecttypes/api/serializers.py

```diff
@@ -1,26 +1,58 @@
 from rest_framework import serializers
+from rest_framework_nested.relations import NestedHyperlinkedRelatedField
+from rest_framework_nested.serializers import NestedHyperlinkedModelSerializer
 
 from objecttypes.core.models import ObjectType, ObjectVersion
 
+from .validators import JsonSchemaValidator, VersionUpdateValidator
+
+
+class ObjectVersionSerializer(NestedHyperlinkedModelSerializer):
+    parent_lookup_kwargs = {"objecttype_uuid": "object_type__uuid"}
 
-class ObjectVersionSerializer(serializers.ModelSerializer):
     class Meta:
         model = ObjectVersion
         fields = (
+            "url",
             "version",
+            "objectType",
             "publicationDate",
             "status",
             "jsonSchema",
         )
         extra_kwargs = {
-            "publicationDate": {"source": "publication_date"},
-            "jsonSchema": {"source": "json_schema"},
+            "url": {"lookup_field": "version"},
+            "version": {"read_only": True},
+            "objectType": {
+                "source": "object_type",
+                "lookup_field": "uuid",
+                "read_only": True,
+            },
+            "publicationDate": {"source": "publication_date", "read_only": True},
+            "jsonSchema": {
+                "source": "json_schema",
+                "validators": [JsonSchemaValidator()],
+            },
             "status": {"read_only": True},
         }
+        validators = [VersionUpdateValidator()]
+
+    def create(self, validated_data):
+        kwargs = self.context["request"].resolver_match.kwargs
+        object_type = ObjectType.objects.get(uuid=kwargs["objecttype_uuid"])
+        validated_data["object_type"] = object_type
+
+        return super().create(validated_data)
 
 
 class ObjectTypeSerializer(serializers.HyperlinkedModelSerializer):
-    versions = ObjectVersionSerializer(many=True)
+    versions = NestedHyperlinkedRelatedField(
```

Comment:
I've just realized that this change breaks compatibility with Objects API validator. But I like that it's consistent with Zaken API now. 
@joeribekker Should we display all versions data here? Or I can make a small change in Objects API

Reply:
I like this as well. I don't see the need to expand this property by default.

Reply:
hmm, we don't have integration tests and we don't even have `docker-compose` with two APIs now. Perhaps we can set up a separate repo with integration tests?

Reply:
I like it, but lets make an issue for now so it won't be missed again.

Reply:
https://github.com/maykinmedia/objects-api/issues/102